### PR TITLE
updated network_api_demo

### LINKED ISSUE #2161: Create network example with pluggable encoders
Change the existing [network example](https://github.com/numenta/nupic/blob/master/examples/network/network_api_demo.py) to create the scalar/date encoders and pass them directly to the multi-encoder that is used in the RecordSensor rather than passing in a dictionary with construction arguments.

Part of #2093

### PR DESCRIPTION
**Pull request by akhilaananthram**
Created the sub-encoders for MultiEncoder separately to make swapping out encoders easier. Fixes #2161

### CODE REVIEW

**Comment by akhilaananthram** [comment]
This is for issue #2093.

**Comment by rhyolight** [comment]
@akhilaananthram To properly link to an issue, update the PR description with:

    Fixes #2093

Thanks.

**Comment by rhyolight** [comment]
This is a fix for #2161, right?

**Comment by akhilaananthram** [comment]
This is for #2093.  It is just the start for it.

File: examples/network/network_api_demo.py [hunk 1]
```diff
@@ -83,25 +83,24 @@
 
 def createEncoder():
   """Create the encoder instance for our test and return it."""
+  consumption_args = {
```
Comment (scottpurdy) [inline examples/network/network_api_demo.py]: just pass these directly in constructor

File: examples/network/network_api_demo.py [hunk 2]
```diff
@@ -83,25 +83,24 @@
...
-          "w": 21,
-      },
-      "timestamp_timeOfDay": {
-          "fieldname": u"timestamp",
-          "name": u"timestamp_timeOfDay",
-          "timeOfDay": (21, 9.5),
-          "type": "DateEncoder",
-      },
-  })
+  encoder.addEncoder(u"consumption", consumption_encoder)
```
Comment (scottpurdy) [inline examples/network/network_api_demo.py]: no need to make these strings unicode

**Comment by scottpurdy** [comment]
done with review, thanks!

**Comment by akhilaananthram** [comment]
@scottpurdy - please review

**Comment by scottpurdy** [comment]
:+1:

**Pull request merged**